docs/ops/deployment/index.md

```diff
@@ -104,6 +104,72 @@ Apache Flink ships with first class support for a number of common deployment ta
   </div>
 </div>
 
+## Deployment Modes
+
+Flink can execute jobs in one of three ways:
+ - in Session Mode, 
+ - in a Per-Job Mode, or
+ - in Application Mode.
+
+ The above modes differ in:
+ - the cluster lifecycle and resource isolation guarantees
+ - whether the application's `main()` is executed on the client or on the cluster.
+
+#### Session Mode
+
+*Session mode* assumes an already running cluster and uses the resources of that cluster to execute any 
+submitted application. Applications executed in the same (session) cluster use, and consequently compete
+for, the same resources. This has the advantage that you do not pay the resource overhead of spinning up
+a full cluster for every submitted job. But, if one of the jobs misbehaves or brings down a Task Manager,
+then all jobs running on that Task Manager will be affected by the failure. This, apart from a negative
+impact on the job that caused the failure, implies a potential massive recovery process with all the 
+restarting jobs accessing the filesystem concurrently and making it unavailable to other services. 
+Additionally, having a single cluster running multiple jobs implies more load for the JobManager, who 
+is responsible for the book-keeping of all the jobs in the cluster.
+
+#### Per-Job Mode
+
+Aiming at providing better resource isolation guarantees, the *Per-Job* mode uses the available cluster manager
+framework (e.g. YARN, Kubernetes) to spin up a cluster for each submitted job. This cluster is available to 
+that job only. When the job finishes, the cluster is torn down and any lingering resources (files, etc) are
+cleared up. This provides better resource isolation, as a misbehaving job can only bring down its own 
+TaskManagers. In addition, it spreads the load of book-keeping across multiple Job Managers, as there is 
```

Comment:
```suggestion
`TaskManagers`. In addition, it spreads the load of book-keeping across multiple Job Managers, as there is 
```

also, I think Job Manager is by now called master or Flink master, see https://ci.apache.org/projects/flink/flink-docs-master/concepts/glossary.html